src/LibHdb++Cassandra.cpp

```
@@ -276,214 +407,197 @@ bool HdbPPCassandra::find_attr_id_and_ttl(string fqdn_attr_name, CassUuid & ID,
-			if(get_ttl_cass_error != CASS_OK)
-			{
+			DEBUG_MSG("Column ttl does not exist???" << endl);
+			// TODO (SJ) Is this an error condition?
+		}
+		else
+		{
+			DEBUG_MSG("OK . value_type() = " << cass_value_type(cassTtlVal) << endl);
+		}
 
-#ifdef _CASS_LIB_DEBUG
-				if(get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
-				{
-					cout << "TTL value is NULL" << endl;
-				}
-				cout << "An error occured: " <<  cass_error_desc(get_ttl_cass_error) << endl;
-				cout << "cass_value_get_int32 for ttl detected a NULL value" << endl;
-#endif
-				ttl = 0;
-			}
-			else
-			{
-#ifdef _CASS_LIB_DEBUG
-				cout << "cass_value_get_int32 returned " << ttl_val << endl;
-#endif
-				ttl = ttl_val;
-			}
-			struct ConfParams conf_param;
-			conf_param.id = ID;
-			conf_param.ttl = ttl;
-#ifdef _CASS_LIB_DEBUG
-			cout << __func__<< "(" << fqdn_attr_name << "): ID found " << endl;
-			cout << __func__<< "(" << fqdn_attr_name << "): TTL = " << ttl << endl;
-#endif
-			found = true;
-			attr_ID_map.insert(make_pair(fqdn_attr_name,conf_param));
-			map<string,struct ConfParams>::iterator it = attr_ID_map.find(fqdn_attr_name);
-			if(it == attr_ID_map.end())
-			{
-				char uuidStr[CASS_UUID_STRING_LENGTH];
-				cass_uuid_string(ID,uuidStr);
-				cout << "WARNING: (" << __FILE__ << "," << __LINE__ << ") "<< __func__<< ": ID (" << uuidStr << ") could not be added into cache for attr ="<< fqdn_attr_name << endl;
-			}
-    	}
-		cass_result_free(result);
-		cass_iterator_free(iterator);
+		CassError get_ttl_cass_error = cass_value_get_int32(cass_row_get_column_by_name(row, CONF_COL_TTL.c_str()), &ttl_val);
+
+		if(get_ttl_cass_error != CASS_OK)
+		{
+			if (get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
+				DEBUG_MSG("TTL value is NULL" << endl);
+
+			DEBUG_MSG("An error occured: " << cass_error_desc(get_ttl_cass_error) << endl);
+			DEBUG_MSG("cass_value_get_int32 for ttl detected a NULL value" << endl);
+			// TODO (SJ) Is this an error condition?
```

This is not an error condition... It could be that the TTL column was simply not set for this attribute (after upgrading from a previous version of the HDB++ cassandra DB schema where this column was not existing) or removed for an unknown reason.
This should be considered as ttl = 0 as it is correctly done in the version you are proposing.
So I think you could remove this TODO and maybe add a comment explaining why this is not an error condition...
I think the DEBUG_MSG message signaling an error should be there only if (get_ttl_cass_error != CASS_ERROR_LIB_NULL_VALUE)
`DEBUG_MSG("cass_value_get_int32 for ttl detected a NULL value" << endl);` could be removed

---

We end up with:

```
		if(get_ttl_cass_error != CASS_OK)
		{
			if (get_ttl_cass_error == CASS_ERROR_LIB_NULL_VALUE)
				DEBUG_MSG("TTL value is NULL" << endl);
			else
				DEBUG_MSG("An error occured: " << cass_error_desc(get_ttl_cass_error) << endl);

			ttl = 0;
		}
````

Is that else an error condition we should raise an exception for? It appears to be an unknown errtor state?

---

Good point... I think indeed an exception should be raised in the else part.

---

Updated to raise an exception